### PR TITLE
CORE-5444: some new persistence service internal tests

### PR DESCRIPTION
**Pull request by dickon**
CORE-5444: some new persistence service internal tests

Refactoring of the test cases, including inlining DbTestContext,
and moving serialization in to helper functions. The goal is to
make the tests themselves concise and readable, which made it
much easier to add more. Also, rename some test methods
for consistency.

### CODE REVIEW

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt [hunk 1]
```diff
@@ -273,229 +253,197 @@ class PersistenceServiceInternalTests {
...
-        val entityResponseSuccess = entityResponse.responseType as EntityResponseSuccess
-        val bytes = entityResponseSuccess.results as List<ByteBuffer>
-        val responseEntity = ctx.deserialize(bytes.first())
-        assertThat(responseEntity).isEqualTo(bellaTheDog)
+    @Test
+    fun `merge multiple in db`() {
+        // save a dog
+        val dog = sandbox.createDog("Basil")
+        val dog2 = sandbox.createDog("Lassie")
+        persistDirectInDb(dog.instance)
```
Comment (mbrkic-r3) [inline components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: dog2 should be persisted as well?

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt [hunk 2]
```diff
@@ -623,72 +562,59 @@ class PersistenceServiceInternalTests {
     /** Cat class has composite key, so also check we find those ok */
     @Test
     fun `find all with composite key`() {
-        val expected = persistCats(ctx, 1)
+        val expected = persistCats()
         val results = assertQuery(QuerySetup.All(CAT_CLASS_NAME))
 
         // Of the expected size - if it fails here - check we're cleaning up elsewhere.
         assertThat(results.size).isGreaterThanOrEqualTo(expected)
```
Comment (mbrkic-r3) [inline components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: should the check here be "equalTo" (and maybe clean-up records here in test setup, if it's difficult to clean up elsewhere)?

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt (hunk 1 above)
Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: In this case, we are testing using merge to both update an entity that does exist in the database, (dog) and that does not exist (dog2). I've added a comment to clarify.

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt (hunk 2 above)
Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: I'll try going to equalTo, and see if that easily works reliably, it's better to be stricter.

Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: equalTo works locally, so I've added it to the branch and we'll see if it works on CI.

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt [hunk 3]
```diff
@@ -515,7 +462,7 @@ class PersistenceServiceInternalTests {
         assertThat(resultsBalance.size).isEqualTo(expected - 4)
 
         // And check the types we've returned
-        val dogClass = ctx.entitySandboxService.getClass(ctx.virtualNodeInfo.holdingIdentity, DOG_CLASS_NAME)
+        val dogClass = entitySandboxService.getClass(virtualNodeInfo.holdingIdentity, DOG_CLASS_NAME)
         results1.forEach {
```
Comment (mbrkic-r3) [inline components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: Code below was not modified with this PR, but might be worth to simplify it:

val allResults = results1 + results2 + resultsBalance
allResults.forEach {
    assertThat(it).isInstanceOf(dogClass)
}
// Check results don't overlap
assertThat(allResults.map { it.toString() }.toSet().size).isEqualTo(expected)

**Review by mbrkic-r3 — DISMISSED** [review]
LGTM

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt (hunk 3 above)
Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: Done.

**Review by mbrkic-r3 — APPROVED** [review]

**Pull request merged**